File: java/src/org/gessinger/gepard/xmp/FileReceiver.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;

import java.util.List ;

public class FileReceiver
{
  static public void main ( String[] args )
  {
    try
    {
      FileReceiver j = new FileReceiver() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  FileReceiver()
  throws Exception
  {
    final Client client = Client.getInstance() ;

    System.out.println ( "Wait for incoming file in a FileContainer." ) ;

    client.on ( "__FILE__", new EventListener()
    {
      public void event ( Event e )
      {
        try
        {
          FileContainer fileContainer = (FileContainer) e.removeValue ( "DATA" ) ;
          String fname = fileContainer.getName() + ".in" ;
          fileContainer.write ( fname ) ;
          System.out.println ( fname + " written." );
          e.setStatus ( 0, "success", "File accepted." ) ;
        }
        catch ( Exception exc )
        {
          e.setStatus ( 1, "error", "File not saved." ) ;
          System.out.println ( Util.toString ( exc ) ) ;
        }
        try
        {
          e.sendBack() ;
        }
        catch ( Exception exc )
        {
          System.out.println ( Util.toString ( exc ) ) ;
        }
      }
    } ) ;
  }
}

File: python/xmp/FileReceiver.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )

import gepard

client = gepard.Client.getInstance()
# client.setDaemon ( True )
def on_close ( err, info ):
	print ( err )
def on_error ( err, info ):
	print ( err )
def on_shutdown ( err, info ):
	print ( "shutdown called" )

client.onClose ( on_close )
client.onError ( on_error )
client.onShutdown ( on_shutdown )

def on___FILE__ ( e ):
	data = e.removeValue ( "DATA" )
	print ( data.getName() + " received." ) ;
	fname = data.getName() + ".in"
	try:
		data.write ( fname ) ;
		print ( fname + " written.")
		e.setStatus ( 0, "success", "File accepted.")
	except Exception as exc:
		print ( exc )
		e.setStatus ( 1, "error", "File not accepted.")
	try:
		e.sendBack() ;
	except Exception as e:
		print ( e )

client.on ( "__FILE__", on___FILE__ )
gepard.util.exitWithSIGINT()
